main-support.ts

```diff
@@ -26,6 +26,8 @@ interface ResolutionMeta {
   bucket: number;
 }
 
+export type ImportStage = 'importingMeta' | 'importingScreenshots' | 'done';
```

Comment:
I'm not a pro at _TypeScript_ -- it seems that the way I handle it makes a short one-liner with a strict requirement on anything typed as `ImportStage` (spell-checking as you type). Enums may be used here too, though I feel like it would require more typing (though it might have auto-complete once you put the `.`) 🤷‍♂ 
https://stackoverflow.com/a/47351677/5017391